_posts/en/2017-03-20-the-polite-code-reviewer.md

```diff
@@ -0,0 +1,202 @@
+---
+layout: default
+title:  "The Polite Code Reviewer"
+date:   2017-03-20 17:50:00
+categories:
+  - en
+  - working-with-others
+---
+
+## The Polite Code Reviewer
+
+### TL;DR
+
+ - Code Review is an opportunity educate,
+but to learn also, and at the same rate.
+ - Don't comment before understanding the reasoning behind the changes.
+ - Spend the same energy recognizing good stuff
+(encouraging and/or praising it)
+that you apply to find bad stuff.
+ - Discussions in code reviews can and should
+a learning material (and serve for future references).
+ - Point clearly and with (code) counter examples
+everything that is not that good.
+ - Recognize the effort that the coleagues employed
+on trying to make the best they can.
+
+## What is this post about, guyz!? OMG!
+
+Lately some work colleagues have said to me
+that my code reviews are a little bit "different".
+Being the very self-confident person that I am,
+I started to ask myself (and other people)
+what could I do to make them better.
+Then I discovered,
+much to my surprise,
+that the "different" in this case is actually a good thing.
+
+I'll try to break down into steps
+the thought process that make me review code
+the way I do nowadays.
+
+### Read the code (_and not "just" the diff_ )
+
+At least for me is very tempting to just glance my eyes over the _diff_.
+And I normally try my best to avoid this urge
+and *really*® read the changes.
+
+This is my way of understand the context for the changes,
+and I think this is pre-condition
+to make up any type of conclusions.
+
+#### The objective here is to bring up the motivations
+
+ - Why something is writen in the way it is?
+ - What was the thought process the developer used here?
+ - Can I learn something from it?
+  (not just technically but even about the business?)
+
+Maybe that piece of ~~shit~~ code
+that appears to be irremediably moronic
+is in there due to some strange historical reasons.
+And the first "code reading" that I do
+aim to do just that:
+get me in a position where the code changes
+are wrapped in a context that fits in my mind.
+
+So to *really*® read the PR before make any comments
+is something that works very well for me (I guess).
+
+### Emphasize the goods before bashing the bads
+
+As alien as that concept may appear I really (want to) believe that
+the main concern of the code reviewer is to emphasize the good stuffs.
+This way those can find the atmosphere to they keep poping up.
+
+IMHO is much more effective to avoid bad coding from "happen"
+than just complain when it happens.
+And no worries or false hopes:
+
+#### _bad code is going "to happen"_
+
+But my advice is: just relax
+and focus the review on appreciation of the good stuffs.
+Again, since this is a difficult thing for us programmers
+(should I dare to say humans?),
+here are some examples that I learnt from time
```

Comment:
Yep, makes sense.
I was trying to saying that time actually taught me this things.
Maybe I should write just that XD